### PR TITLE
Remove _write overrides

### PR DESCRIPTION
**Pull request by xmo-odoo**
Task 2186312

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
Hello,

Could you make it tested by Sbu for the slides part (just ping him when your branch is ready) ?

Many thanks !

**Comment by dbo-odoo** [comment]
@tivisse is now doing `sale`, so i gave him my review request as a gift :gift: :heart_decoration:

**Comment by xmo-odoo** [comment]
@dbo-odoo @tivisse already had to review changes in an other modules, you're going to kill him :(

File: addons/sale/models/sale.py [hunk 1]
```diff
@@ -476,31 +476,23 @@ def create(self, vals):
...
         return result
 
-    def _write(self, values):
-        """ Override of private write method in order to generate activities
-        based in the invoice status. As the invoice status is a computed field
-        triggered notably when its lines and linked invoice status changes the
-        flow does not necessarily goes through write if the action was not done
-        on the SO itself. We hence override the _write to catch the computation
-        of invoice_status field. """
-        if self.env.context.get('mail_activity_automation_skip'):
```
Comment (dbo-odoo) [inline addons/sale/models/sale.py:486]: you've lost the skip based on the context key here

File: addons/sale/models/sale.py [hunk 2]
```diff
@@ -476,31 +476,23 @@ def create(self, vals):
...
-                        'sale.mail_act_sale_upsell',
-                        user_id=order.user_id.id,
-                        note=_("Upsell <a href='#' data-oe-model='%s' data-oe-id='%d'>%s</a> for customer <a href='#' data-oe-model='%s' data-oe-id='%s'>%s</a>") % (
-                            order._name, order.id, order.name,
-                            order.partner_id._name, order.partner_id.id, order.partner_id.display_name))
-
-        return super(SaleOrder, self)._write(values)
+    def _compute_field_value(self, field):
+        super()._compute_field_value(field)
+        if field.name != 'invoice_status':
```
Comment (xmo-odoo) [inline addons/sale/models/sale.py:481]:
```suggestion
        if field.name != 'invoice_status' or self.env.context.get('mail_activity_automation_skip'):
```

File: addons/sale/models/sale.py (hunk 1 above)
Comment (xmo-odoo) [reply addons/sale/models/sale.py:486]: tru, put it back, though it's really just an optimisation (both `activity_unlink` and `activity_schedule` check for it).

File: addons/sale/models/sale.py (hunk 2 above)
Comment (dbo-odoo) [reply addons/sale/models/sale.py]: if it's checked in the subsequent activity calls then no change to your pr is needed i should think, feel free to add this suggestion or not based on your current mood

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by xmo-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**